proposal: add build tag to exclude example tools from `go install ./...`

Running `go install ./...` on VGE currently results in build errors for the example directory, as there exist multiple Go source files defining the `main` function in the same directory.

```
$ go install -v ./...
github.com/lakal3/vge/vge/materials/phong
github.com/lakal3/vge/examples/basic
# github.com/lakal3/vge/examples/basic
examples/basic/logo.go:16:6: main redeclared in this block
	previous declaration at examples/basic/hello.go:13:6
examples/basic/logo.go:31:6: buildScene redeclared in this block
	previous declaration at examples/basic/hello.go:29:21
examples/basic/logo_an.go:21:6: main redeclared in this block
	previous declaration at examples/basic/logo.go:16:6
examples/basic/logo_an.go:50:6: buildScene redeclared in this block
	previous declaration at examples/basic/logo.go:31:21
examples/basic/logo_an.go:152:6: nodeVisible redeclared in this block
	previous declaration at examples/basic/logo.go:109:6
examples/basic/logo_an.go:156:23: (*nodeVisible).Process redeclared in this block
	previous declaration at examples/basic/logo.go:113:6
github.com/lakal3/vge/examples/model
```

To resolve this issue, one approach is to add build tags (Go proposal for new build tag syntax https://github.com/golang/go/issues/41184) which exclude these examples from default builds, e.g.

```go
//go:build examples
```

--- a/examples/basic/hello.go
+++ b/examples/basic/hello.go
@@ -1,3 +1,5 @@
+//+build examples
+
 package main
 
 import (

--- a/examples/basic/logo.go
+++ b/examples/basic/logo.go
@@ -1,3 +1,5 @@
+//+build examples
+
 package main
 
 import (

--- a/examples/basic/logo_an.go
+++ b/examples/basic/logo_an.go
@@ -1,3 +1,5 @@
+//+build examples
+
 package main
 
 import (

